src/libmeasurement_kit/dns/system_resolver_impl.hpp

```diff
@@ -1,194 +0,0 @@
-    if (error) {
-        Error resolver_error;
-        switch (error) {
-        case EAI_AGAIN:
-            resolver_error = TemporaryFailureError();
-            break;
-        case EAI_BADFLAGS:
-            resolver_error = InvalidFlagsValueError();
-            break;
-#ifdef EAI_BADHINTS  // Not always available
-        case EAI_BADHINTS:
-            resolver_error = InvalidHintsValueError();
-            break;
-#endif
-        case EAI_FAIL:
-            resolver_error = NonRecoverableFailureError();
-            break;
-        case EAI_FAMILY:
-            resolver_error = NotSupportedAIFamilyError();
-            break;
-        case EAI_MEMORY:
-            resolver_error = MemoryAllocationFailureError();
-            break;
-        case EAI_NONAME:
-            resolver_error = HostOrServiceNotProvidedOrNotKnownError();
-            break;
-        case EAI_OVERFLOW:
-            resolver_error = ArgumentBufferOverflowError();
-            break;
-#ifdef EAI_PROTOCOL  // Not always available
-        case EAI_PROTOCOL:
-            resolver_error = UnknownResolvedProtocolError();
-            break;
-#endif
-        case EAI_SERVICE:
-            resolver_error = NotSupportedServnameError();
-            break;
-        case EAI_SOCKTYPE:
-            resolver_error = NotSupportedAISocktypeError();
-            break;
-        default:
-            resolver_error = ResolverError();
-            break;
-        }
-        ctx->reactor->call_soon([=]() { callback(resolver_error, nullptr); });
-        return;
-    }
-    assert(sip != nullptr);
-    addrinfo_uptr servinfo(sip);
-
-    void *addr_ptr;
-    char abuf[128];
-    for (addrinfo *p = servinfo.get(); p != nullptr; p = p->ai_next) {
-        Answer answer;
-        answer.name = ctx->name;
-        answer.qclass = ctx->dns_class;
-        assert(p->ai_family == AF_INET or p->ai_family == AF_INET6);
-        if (p->ai_family == AF_INET) {
-            answer.type = MK_DNS_TYPE_A;
-            addr_ptr = &((sockaddr_in *)p->ai_addr)->sin_addr;
-        } else if (p->ai_family == AF_INET6) {
-            answer.type = MK_DNS_TYPE_AAAA;
-            addr_ptr = &((sockaddr_in6 *)p->ai_addr)->sin6_addr;
-        } else {
-            // Added branch to avoid g++ compiler warning
-            throw std::runtime_error("internal error");
-        }
-        if (p->ai_canonname != nullptr) {
-            answer.hostname = p->ai_canonname;
-        }
-        if (inet_ntop(p->ai_family, addr_ptr, abuf, sizeof(abuf)) == nullptr) {
-            ctx->logger->warn("dns: unexpected inet_ntop failure");
-            ctx->reactor->call_soon(
-                [=]() { callback(InetNtopFailureError(), nullptr); });
-            return;
-        }
-        if (p->ai_family == AF_INET) {
-            answer.ipv4 = std::string(abuf);
-        } else if (p->ai_family == AF_INET6) {
-            answer.ipv6 = std::string(abuf);
-        }
-        message->answers.push_back(answer);
-    }
```

Comment:
This code to process result is now a specific function in `getaddrinfo_async.hpp` that just does that task.